test/release/examples/patterns/readcsv.chpl

```diff
@@ -0,0 +1,286 @@
+  // Note: this portion is the same as in Approach 1
+  var line : string;
+
+  if (!reader.readline(line)) then
+    writeln("ERROR: ", inFileName, " appears to be empty");
+
+  var colNames : list(string);
+  colNames = createListOfColNames(line);
+
+  if debug {
+    writeln("\ncolNames: ", colNames);
+  }
+
+  // Create a list of lines from the file, so we know how many lines of data there are.
+  // Then create an associative array with an entry per column of data, with the
+  // associative key being the column name.  The value will be an array of strings
+  // with the column's value for each row in the data set.
+  var dataRows : list(string);
+
+  // Reading all of the lines of the file into a list.
+  while (reader.readline(line)) {
+    dataRows.append(line);
+  }
+
+  // Declaring an associative array, where the value type is a 1D array large enough
+  // to store one column value per row of data.
+  var valDomain = {0..dataRows.size-1}; // Domain indicating how many values per column.
+  var colNameDomain: domain(string);    // An associative domain of strings.
+  for colName in colNames do            // Put each column name into the column name domain.
+    colNameDomain += colName;
+  // colData is an associative array indexed by the column name domain with array values.
+  // Each array value will have number of rows string elements.
+  var colData: [colNameDomain] [valDomain] string;
+
+  // Processing each row to put the column values into the associative array.
+  for rowIdx in 0..dataRows.size-1 {
+    // Pull the line out from the list of dataRows.
+    line = dataRows[rowIdx];
+    var start = 0;      // index into the line string
+    var colIdx = 0;     // the first column is indexed as zero
+    // Process each line by finding the next field value from the start index to the next comma.
+    while (start<line.size && colIdx<colNames.size) {
+      var (nextVal,nextCommaIdx) = nextField(line,start);
+      colData[colNames[colIdx]][rowIdx] = nextVal;
+      start = nextCommaIdx + 1;         // The next column of data will start after the comma.
+      colIdx = colIdx + 1;              // Increment the column index.
+    }
+  }
+  reader.close();
+
+  // Print out all of the data values for the first column of data.
+  if debug then writeln("\ncolData[colNames[0]]=",colData[colNames[0]]);
+
+  //-----------------------------------------------------------------------
+  // To see how the data is accessed and for testing, let's generate a 
+  // temporary csv file from the stored data.  It should be the same as the 
+  // input csv file.
+  if debug {
+    var outfile = open(approach2FileName, iomode.cw);
+    var writer = outfile.writer();
+
+    // First write to the output file the column names separated by commas.
+    for colIdx in 0..colNames.size-2 {
+      writer.write(colNames[colIdx],",");
+    }
+    // The last column name should not be followed by comma.
+    writer.writeln(colNames[colNames.size-1]);  
```

Comment:
Similar one-liner, or helper. Another example is below

Reply:
Not going to change this one.